### PR TITLE
Disable git fsmonitor in git_worker to fix hangs when fetching repositories

### PR DESCRIPTION
**Pull request by luispadron**
This commit adds the `-c core.fsmonitor=false` arguments to all git commands (excluding `git init`) in `git_worker.bzl`. The purpose of this is to disable git from spawning a file system monitor, when many of these are spawned on a machine that is already under load or out of file descriptors, the git commands will hang indefinitely.

Fixes https://github.com/bazelbuild/bazel/issues/21438

### CODE REVIEW

**Comment by luispadron** [comment]
I performed tests by spawning many file watchers using the `fswatch` tool, using a released Bazel version `7.3.1` when running `bazel fetch` in the `examples/lots_of_repos` directory of the branch: https://github.com/cgrindel/rules_swift_package_manager/tree/luis/lots-of-repo-example I was able to consistently reproduce stuck fetches. 

I confirmed that building `//src:bazel` and using that instead consistently allows all fetches to continue and I ran into no hangs during the process.

---

```sh
#!/bin/bash

# Number of monitors to create
NUM_MONITORS=100  # Adjust based on your stress test needs

# Directory to monitor (can be the same or different for each monitor)
MONITOR_DIR="/tmp/fsmonitor-test"

# Create the directory if it doesn't exist
mkdir -p "$MONITOR_DIR"

# Array to store process IDs of background fswatch processes
declare -a PIDS

# Function to create and run a file system monitor
create_monitor() {
  local dir=$1
  fswatch "$dir" >/dev/null &
  PIDS+=($!)  # Store the process ID
}

# Create the specified number of monitors
for i in $(seq 1 $NUM_MONITORS); do
  create_monitor "$MONITOR_DIR"
  
  # Optionally, introduce a slight delay to avoid overwhelming the system too quickly
  sleep 0.1
done

echo "Created $NUM_MONITORS file system monitors."


# Function to clean up by killing all background processes
cleanup() {
  echo "Cleaning up monitors..."
  for pid in "${PIDS[@]}"; do
    kill "$pid" 2>/dev/null
  done
  echo "All monitors terminated."
}

# Trap script exit (Ctrl+C or normal exit) to ensure cleanup happens
trap cleanup EXIT

# Keep the script running to simulate long-running monitors
echo "Press Ctrl+C to stop the script and clean up the monitors."
while true; do
  sleep 10  # Keep the script alive indefinitely
done

File: tools/build_defs/repo/git_worker.bzl [hunk 1]
```diff
@@ -196,8 +196,11 @@ def _git_maybe_shallow(ctx, git_repo, command, *args):
     return _execute(ctx, git_repo, start + args_list)
 
 def _execute(ctx, git_repo, args):
```
Comment (luispadron) [inline tools/build_defs/repo/git_worker.bzl:198]: I moved the passing of `git` to be implied in this function as its intended to run `git` commands anyway. This allows me to add `-c` which must come directly after `git` without searching through the `args` we are given

**Comment by fmeum** [comment]
@bazel-io fork 7.4.0

**Review by Wyverald — APPROVED** [review]
nice! thanks for providing lots of context too, that really helps.

**Pull request closed without merging**